storagemarket/impl/providerstates/provider_states.go

```diff
@@ -341,17 +345,26 @@ func HandoffDeal(ctx fsm.Context, environment ProviderDealEnvironment, deal stor
 		_ = ctx.Trigger(storagemarket.ProviderEventPieceStoreErrored, err)
 	}
 
-	if err := environment.ActivateShard(deal.Proposal.PieceCID); err != nil {
+	if err := environment.RegisterShard(ctx.Context(), deal.Proposal.PieceCID, carFilePath); err != nil {
 		// TODO What's the right thing to do here ? I think the retrieval market
 		// should have a recovery mechanism in terms of, let "activate the shard if you don't have it".
+		err = xerrors.Errorf("failed to activate shard: %w", err)
+		log.Error(err)
+		//return ctx.Trigger(storagemarket.ProviderEventDealHandoffFailed, err)
 	}
 
-	// TODO Put code in Lotus to expire/destory these shards when deals expire.
+	// TODO Put code in Lotus to expire/destroy these shards when deals expire.
```

Comment:
We can remove this TODO now as it has been extracted as a separate work item and will now be done in Lotus.